File: ms-usuarios/src/main/java/com/sylvie/usuarios/entity/Usuario.java
package com.sylvie.usuarios.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "usuario", schema = "usuarios")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombre;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String password;

    @Column(name = "fecha_registro")
    private LocalDateTime fechaRegistro = LocalDateTime.now();
}

File: sylvie-app/app/src/main/java/com/sylvie/app/data/models/Usuario.kt
package com.sylvie.app.data.models

data class Usuario(
    val id: Long,
    val nombre: String,
    val email: String
)

data class RegistroRequest(
    val nombre: String,
    val email: String,
    val password: String
)

data class LoginRequest(
    val email: String,
    val password: String
)

data class RestriccionRequest(
    val ingrediente: String,
    val tipo: String
)